File: app/src/main/java/com/vikingsen/cheesedemo/model/data/comment/CommentChange.java
package com.vikingsen.cheesedemo.model.data.comment;


public class CommentChange {
    private final long cheeseId;
    private final boolean bulkChange;

    static CommentChange bulkOperation() {
        return new CommentChange(-1, true);
    }

    static CommentChange forCheese(long cheeseId) {
        return new CommentChange(cheeseId, false);
    }

    private CommentChange(long cheeseId, boolean bulkChange) {
        this.cheeseId = cheeseId;
        this.bulkChange = bulkChange;
    }

    public long getCheeseId() {
        return cheeseId;
    }

    public boolean isBulkChange() {
        return bulkChange;
    }
}

File: app/src/main/java/com/vikingsen/cheesedemo/model/data/comment/CommentChange.kt
package com.vikingsen.cheesedemo.model.data.comment


class CommentChange private constructor(val cheeseId: Long, val isBulkChange: Boolean) {
    companion object {
        fun bulkOperation(): CommentChange {
            return CommentChange(-1, true)
        }

        fun forCheese(cheeseId: Long): CommentChange {
            return CommentChange(cheeseId, false)
        }
    }
}
